crates/bevy_math/src/linear_convex_space.rs

```diff
@@ -0,0 +1,77 @@
+//! This module provides the `LinearConvexSpace` trait and some related objects
+use glam::{DVec2, DVec3, DVec4, Quat, UVec2, UVec3, UVec4, Vec2, Vec3, Vec3A, Vec4};
+
+/// A trait providing methods for spanning a linear, convex space.
+///
+/// These are a generally useful set of properties, but in particular, are needed to ensure spline interpolation is well-behaved.
+/// A convex space or convex set is a set of points (think points in 3d or 2d space) in which every point on a line between two points A and B in that set
+/// is also a part of that set. For example the set of all points inside a square is a convex set as you can draw a line between any two points in the square and never leave it.
+/// For more information, please see [this wikipedia article](https://en.wikipedia.org/wiki/Convex_set).
+///
+/// The space must also be linear,  allowing you to interpolate between any two points `A` and `B` via the parameter
```

Comment:
Math nitpick:
This is not exactly what linearity means, as it goes backwards, a line is usually defined using operations on a linear space.

The properties that actually make for a Linear space (also known as a Vector space) are:

(`a` and `b` are arbitrary scalars, `x` and `y` arbitrary vectors.)
A) Addition is commutative
B) Addition is associative
C) `0 * x` produces an identity element (`0`).
D) `1 * x = x`
E) Scalar multiplication associates with number multiplication `(ab)*x = a*(b*x)`
F) Scalar multiplication distributes over addition `(a+b)*(x+y) = a*x + a*y + b*x + b*y`

(I hope I didn't miss anything.)

These properties make up a vector space, which is what API consumers expect. (I believe these properties could form their own trait and convexity would be a separate marker trait. But that's maybe a next-PR task.)

As far as convexity goes, you're right, although we could note that the `line between points A and B` is defined as all the points `A*t + B*(1-t)` because it might not be imideatly obvious what is a line segment in an abstract space.

Reply:
Yep, I think I'd prefer to include this style of definition.

Reply:
Could someone write this as a suggestion please? I am not a native speaker and I don't think I could write good docs for what is described above although I agree with the spirit. :/